### PR TITLE
Robin/eng 2136 add lambdalog

### PR DESCRIPTION
**Pull request by Latnac**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,18 +15,14 @@
...
     "logger",
-    "mos",
-    "winston"
+    "mos"
   ],
   "author": "David Zhang",
   "dependencies": {
-    "sprintf-js": "^1.1.2",
-    "winston": "^3.7.2"
+    "lambda-log": "^3.1.0"
```
Comment (antoinewg) [inline package.json:22]: + `npm i -D @types/lambda-log`

File: package.json [hunk 2]
```diff
@@ -15,18 +15,14 @@
...
   "keywords": [
     "logger",
-    "mos",
-    "winston"
+    "mos"
   ],
   "author": "David Zhang",
   "dependencies": {
-    "sprintf-js": "^1.1.2",
-    "winston": "^3.7.2"
```
Comment (antoinewg) [inline package.json:24]: 🎉

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { LambdaLog } from 'lambda-log'
+
+export default new LambdaLog({
+    dev: process.env.stage === 'dev',
+    stage: process.env.stage,
```
Comment (antoinewg) [inline src/index.ts:5]: this option is invalid. Maybe in `meta` or `dynamicMeta` ?

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    dynamicMeta() {
+        return {
+            timestamp: new Date().toISOString()
```
Comment (antoinewg) [inline src/index.ts:8]: with this interface we can just import a logger and use it, we can dynamically change options from within the consumer. For example:

```typescript
// in apischolarship for instance
import logger from 'logger'
logger("hello")...
// we can't add a tag without import lambda-log
```

What about ‘exposing' the interface of lambda log like this:

```ts
const Logger = (options?: LambdaLogOptions) => {
    const { meta = {}, dynamicMeta = () => {}, ...otherOptions } = options || {}

    return new LambdaLog({
        dev: process.env.stage === 'dev',
        meta: {
            stage: process.env.stage,
            ...meta,
        },
        dynamicMeta: (message) => ({
            timestamp: new Date().toISOString(),
            ...dynamicMeta(message),
        }),
        ...otherOptions,
    })
}
```

this way we can:

```ts
import Logger from "logger"
const logger = Logger({meta: { service: "apischolarship" }})
```

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { LambdaLog } from 'lambda-log'
+
+export default new LambdaLog({
+    dev: process.env.stage === 'dev',
```
Comment (antoinewg) [inline src/index.ts:4]: why not keep it `development` like before ?

Comment (antoinewg) [reply src/index.ts:4]: Also, it's `NODE_ENV` right ?

File: package.json (hunk 2 above)
Comment (antoinewg) [reply package.json:24]: That makes it sooo much lighter

image

image

File: package.json (hunk 1 above)
Comment (Latnac) [reply package.json:22]: @types/lambda-log is version 2, so it is showing errors that should not be there. For example an error if you try to log an error directly 
- log.info(err)
Which is perfectly processed by lambda-log v3
So I prefer to not add it

**Pull request merged**